src/MySqlConnector/Serialization/MySqlSession.cs

```diff
@@ -301,6 +301,55 @@ public void FinishQuerying()
 				await SendReplyAsync(payload, ioBehavior, cancellationToken).ConfigureAwait(false);
 				break;
 
+			case "sha256_password":
+				// add NUL terminator to password
+				var passwordBytes = Encoding.UTF8.GetBytes(password);
+				Array.Resize(ref passwordBytes, passwordBytes.Length + 1);
+
+				if (!m_isSecureConnection && passwordBytes.Length > 1)
+				{
+#if NET451
+					throw new MySqlException("Authentication method '{0}' requires a secure connection (prior to .NET 4.6).".FormatInvariant(switchRequest.Name));
+#else
+					// request the RSA public key
```

Comment:
In a future PR, we should add support for including the Server's Public Key File in a command line argument to save this round trip